disco_aws_automation/disco_bake.py

```diff
@@ -445,12 +445,12 @@ def bake_ami(self, hostclass, no_destroy, source_ami_id=None, stage=None):
         return image
 
     @staticmethod
-    def _tag_ami_with_metadata(ami, stage, source_ami_id, productline=None):
+    def _tag_ami_with_metadata(ami, stage, source_ami_id, productline=None, extra_tags=None):
         """
         Tags an AMI with the stage, source_ami, the branch/git-hash of disco_aws_automation,
         and the productline if provided
         """
-        tag_dict = OrderedDict()
+        tag_dict = extra_tags or OrderedDict()
```

Comment:
This is what happens when we delete comments for some reason. Looks like the reason we had an OrderedDict was because we use to put a lot of tags on AMIs, and we wanted to make sure that the tags that Asiaq needs would always be the first ones applied to the AMI. See https://github.com/amplifylitco/X_DEPRECATED_asiaq/commit/b90a9aec2ada04d7ca9069150cc2fcdb0d846ceb#diff-cdc32122df064b229e21a9c4e04fc2c1R878

This seems like less of a requirement now that AWS allows us to set 50 tags per resource, but someone could still try to set 50 tags on the command line, so might as well continue the same behavior.